src/js/main.js

```diff
@@ -64,41 +64,95 @@ Site.Menu = {
 };
 
 Site.Shapes = {
-  currentState: 0,
-  maxState: 7,
+  $mapPattern: $('.map-pattern'),
+  $patterns: [
+    $('#pattern-1'),
+    $('#pattern-2'),
+    $('#pattern-3'),
+    $('#pattern-4'),
+    $('#pattern-5'),
+  ],
+  patternMin: 0,
+  patternMax: 4,
+  currentPattern: 0,
   timer: null,
-  interval: 12000,
+  interval: 100,
+  animating: false,
+
   init: function() {
     var _this = this;
 
-    _this.setState();
-    _this.startInterval();
+    _this.showPattern();
+
+    document.querySelector('#map').addEventListener('startpan', function() {
```

Comment:
obv this code is fine and correct so this comment is about jquery vs vanilla convo: 

yh I don't really see the point in not writing jquery. this is just long and not as clean as a nice jquery `.on` with an event map. the perf issues dont really exist in this example as the querySelector has to run twice (could be cached tho) and then bind to the event. if this query was happening a lot then maybe the perf bump would have some effect.

the jquery equivalent is:

```
$('#map').on({
  'startpan': function() {...},
  'stoppan': function() {...},
});
```

which I def prefer